File: game-engine/src/main/java/com/animousen4/game/engine/core/repositories/entities/UserEntity.java
package com.animousen4.game.engine.core.repositories.entities;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;
import lombok.experimental.WithBy;

import java.sql.Timestamp;

@Entity
@Getter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@With
@Table(name = "users")
public class UserEntity {
    @Id
    @SequenceGenerator(name = "users_id_seq", sequenceName = "users_id_seq", allocationSize = 1)
    @GeneratedValue(strategy=GenerationType.SEQUENCE, generator = "users_id_seq")
    Long id;

    String username;

    @Column(name = "registration_date")
    Timestamp registrationDate;

    @Column(name = "close_date")
    Timestamp closeDate;

    String email;

    @Column(name = "status_id")
    Long statusId;

    @Column(name = "status_reason_id")
    Long statusReasonId;

}

File: auth-server-webflux/src/main/kotlin/com/animousen4/authserverwebflux/core/entities/UserEntity.kt
package com.animousen4.authserverwebflux.core.entities

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import java.util.Date


@Table
data class UserEntity (
    @Id
    val id: Long,
    val username: String,
    val registrationDate: Date,
    val closeDate: Date,
    val email: String,
    val statusId: Long,
    val statusReasonId: Long,
    val creatorId: Long
)